workspaces/marketplace/plugins/marketplace-backend/src/router.ts

```diff
@@ -108,6 +159,112 @@ export async function createRouter({
     res.json(plugin);
   });
 
+  router.get(
+    '/plugin/:namespace/:name/configuration/authorize',
+    async (req, res) => {
+      const [readDecision, installDecision] = await Promise.all([
+        authorizeConditional(req, extensionPluginReadPermission),
+        authorizeConditional(req, extensionPluginCreatePermission),
+      ]);
+      if (
+        readDecision.result === AuthorizeResult.DENY &&
+        installDecision.result === AuthorizeResult.DENY
+      ) {
+        res.status(403);
+        return;
+      }
+
+      const authorizedActions: string[] = [];
+      let plugin: MarketplacePlugin;
+
+      const evaluateConditional = async (
+        decision: PolicyDecision,
+        action: string,
+      ) => {
+        if (decision.result === AuthorizeResult.CONDITIONAL) {
+          if (!plugin) {
+            plugin = await marketplaceApi.getPluginByName(
+              req.params.namespace,
+              req.params.name,
+            );
+          }
+          if (matches(plugin, decision.conditions)) {
+            authorizedActions.push(action);
+          }
+        } else if (decision.result === AuthorizeResult.ALLOW) {
+          authorizedActions.push(action);
+        }
+      };
+
+      await Promise.all([
+        evaluateConditional(readDecision, 'read'),
+        evaluateConditional(installDecision, 'create'),
+      ]);
+
+      if (authorizedActions.length === 0) {
+        res.status(403);
+        return;
+      }
+      res.status(200).json({ authorizedActions });
+    },
+  );
+
+  router.get('/plugin/:namespace/:name/configuration', async (req, res) => {
+    const readDecision = await authorizeConditional(
+      req,
+      extensionPluginReadPermission,
+    );
+    if (readDecision.result === AuthorizeResult.DENY) {
+      res.status(403).json({ error: 'Forbidden' });
+      return;
```

Comment:
Can we throw an error instead? For all 403 cases.

```tsx
import {
  NotAllowedError,
} from '@backstage/errors';
```

```suggestion
        throw new NotAllowedError(`Not allowed to configure ${req.params.namespace}:${req.params.name}`);
```